File: Typescript/projects/linql.core/src/lib/LinqlConstant.ts
import { LinqlExpression } from "./LinqlExpression";
import { LinqlType } from "./LinqlType";

export class LinqlConstant extends LinqlExpression
{
    "$type": string = "LinqlConstant";

    ConstantType: LinqlType;

    Value: any;

    constructor(ConstantType: LinqlType, Value: any)
    {
        super();
        this.ConstantType = ConstantType;
        this.Value = Value;
    }

    public Clone(): this
    {
        const constant = new LinqlConstant(this.ConstantType, this.Value);
        constant.Next = this.Next?.Clone();
        return constant as this;
    }
}

File: Python/linql-core/src/linql_core/LinqlConstant.py
from typing_extensions import Self
from .LinqlExpression import LinqlExpression
from .LinqlType import LinqlType
from typing import Any

class LinqlConstant(LinqlExpression):
    
    type: str = "LinqlConstant"
    ConstantType: LinqlType
    Value: Any

    def __init__(self, ConstantType: LinqlType, Value: Any):
        self.ConstantType = ConstantType
        self.Value = Value
    
    def Clone(self) -> Self:
        constant = LinqlConstant(self.ConstantType, self.Value)

        if self.Next != None:
            constant.Next = self.Next.Clone()
        return constant
    
    def toSerializable(self) -> dict:
        jsonObject = self._CreateSerializableType()
        
        jsonObject["ConstantType"] = self.ConstantType.toSerializable()

        if hasattr(self, "Value") and self.Value != None:
            jsonObject["Value"] = self.Value

        self._SerializeNext(jsonObject)
        return jsonObject